src/nvim/tui/tui.c

```diff
@@ -69,6 +70,9 @@ typedef struct {
   bool can_use_terminal_scroll;
   bool mouse_enabled;
   bool busy;
+  /* use int instead ? */
+  // cursorentry_T cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
+  int cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
```

Comment:
Oh. No, please don't do that. The TUI follows the same protocol as _all_ UIs. It cannot and must not receive different data than some other UI. Don't worry about saving bytes, especially because this is a rare event.